### ISSUE TITLE
splinter_test --perf runs into Assertion from clockcache.c:2131:clockcache_get_internal(): "allocator_get_ref(cc->al, base_addr) > 1". 

### ISSUE THREAD

**Issue by gapisback** [vmware/splinterdb#305]
Following test run was seen to run into an assertion twice; once in a debug-gcc run and once in an ASAN run on CI.

```
+ bin/driver_test splinter_test --perf --max-async-inflight 0 --num-insert-threads 4 --num-lookup-threads 4 --num-range-lookup-threads 0 --tree-size-gib 2 --cache-capacity-mib 512
bin/driver_test: splinterdb_build_version d7643d50
Dispatch test splinter_test
fingerprint_size: 27
Running splinter_test with 1 caches
splinter_test: splinter performance test started with 1                tables
inserting  74% complete for table 0Assertion failed at src/clockcache.c:2131:clockcache_get_internal(): "allocator_get_ref(cc->al, base_addr) > 1". 
./test.sh: line 41:   279 Aborted                 "$@"
```

See these internal slack threads [here, 2/22/2022](https://vmware.slack.com/archives/G01MEDQ05MX/p1645550951000009) and [here, 3/2022](https://vmware.slack.com/archives/G01MEDQ05MX/p1646207005770419), where this failure was reported and discussed.

After the 1st instance, there was some speculation that this may have been due to `make` and incomplete build issues. 
Some fixes were applied to `Makefile`, addressing rules for unit-tests, but this failure emanates from `bin/driver_test`, so those fixes are not applicable.

Also, the three recent failures were all seen on CI-builds, so there should have been no issue about a `make clean` build etc.

**Comment by gapisback** [comment]
**Weirdly .. this recurred** on CI-jobs in branch [agurajada/341-PgExt-size-limits](https://github.com/vmware/splinterdb/tree/agurajada/341-PgExt-size-limits), for a completely unrelated change-set.

Rebased to /main and [here's the failing CI-job](https://runway-ci.eng.vmware.com/builds/9789947): 

```
+ bin/driver_test splinter_test --perf --max-async-inflight 0 --num-insert-threads 4 --num-lookup-threads 4 --num-range-lookup-threads 0 --tree-size-gib 2 --cache-capacity-mib 512
bin/driver_test: splinterdb_build_version 30cc7439
Dispatch test splinter_test
Fingerprint size 29 too large, max value size is 5, setting to 27
fingerprint_size: 27
filter-index-size: 256 is too small, setting to 512
Running splinter_test with 1 caches
splinter_test: SplinterDB performance test started with 1 tables
inserting  39% complete for table 0 ... Assertion failed at src/clockcache.c:2131:clockcache_get_internal(): "allocator_get_ref(cc->al, base_addr) > 1". 
./test.sh: line 66:   203 Aborted                 "$@"
make: *** [Makefile:312: run-tests] Error 134
```
